lib/dotcom_web/live/trip_planner.ex

```diff
@@ -303,6 +311,35 @@ defmodule DotcomWeb.Live.TripPlanner do
     {:noreply, socket}
   end
 
+  @impl true
+  # We have to handle the result of the push_patch, but we ignore it.
+  def handle_params(_params, _uri, socket) do
+    {:noreply, socket}
+  end
+
+  # When the datetime_type is "leave_at" or "arrive_by", we need to
+  # have a "datetime" indicating when we want to "leave at" or "arrive
+  # by", but because the datepicker only appears after a rider clicks
+  # on "Leave at" or "Arrive by", the actual value of "datetime"
+  # doesn't always appear in params. When that happens, we want to set
+  # "datetime" to a reasonable default.
+  defp add_datetime_if_needed(%{"datetime_type" => "now"} = params),
+    do: params |> Map.put("datetime", Timex.now("America/New_York"))
+
+  defp add_datetime_if_needed(%{"datetime" => datetime} = params) when datetime != nil, do: params
+  defp add_datetime_if_needed(params), do: params |> Map.put("datetime", nearest_5_minutes())
+
+  # Converts a changeset to a map of params that can be encoded into a URL.
+  defp changeset_to_params(%Ecto.Changeset{changes: changes}) do
+    changes
+    |> Map.update(:from, %{}, &Map.get(&1, :changes))
+    |> Map.update(:to, %{}, &Map.get(&1, :changes))
+    |> Map.update(:modes, %{}, &Map.get(&1, :changes))
+    |> Map.new(fn {k, v} -> {to_string(k), v} end)
+    |> Jason.encode!()
+    |> Jason.decode!()
```

Comment:
what is this a workaround for, i'm quite curious

Reply:
this is because the params come in with strings as keys but the changeset has them as atoms.